accounts/views.py

```diff
@@ -0,0 +1,142 @@
+        user.save()
+
+        try:
+            send_email(email, password)
+        except:
+            pass
+
+        return Response({'email': email, 'id': user.id, 'auth_token': token}, status=status.HTTP_201_CREATED)
+
+
+class LoginAPIView(APIView):
+    serializer_class = LoginSerializer
+
+    def post(self, request):
+        """
+        Checks is user exists.
+        Email and password are required.
+        Returns a JSON web token.
+        """
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        email = serializer.validated_data['email']
+        token = serializer.validated_data['auth_token'].key
+
+        user = MyUser.objects.filter(email__exact=email).first()
+
+        return Response({
+            'email': email,
+            'last_name': user.last_name,
+            'first_name': user.first_name,
+            'id': user.id,
+            'auth_token': token
+        }, status=status.HTTP_200_OK)
+
+
+class UserDetailsAPIView(APIView):
+    def get(self, request):
+        """
+        Checks is user exists.
+        Auth Token is required.
+        Returns Email, First name, Last name, Id.
+        """
+        if not request.auth:
+            return Response({'error': 'Auth data is required'}, status=status.HTTP_400_BAD_REQUEST)
+
+        if not request.auth.key:
+            return Response({'error': 'Auth token is required'}, status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            user = Token.objects.get(key=request.auth.key).user
+        except Token.DoesNotExist:
+            return Response({'error': 'User does not exists'}, status=status.HTTP_400_BAD_REQUEST)
+
+        if user is not None:
+            return Response(status=200, data={
+                "id": user.id,
+                "email": user.email,
+                "first_name": user.first_name,
+                "last_name": user.last_name,
+            })
+        else:
+            return Response({'error': 'User does not exists'}, status=status.HTTP_400_BAD_REQUEST)
+
+    def post(self, request):
+        """
+        Checks is user exists.
+        First name, Last name, Auth Token are required.
+        Update First name and Last name.
+        Returns Email, First name, Last name, Id.
+        """
+        data = request.data
+
+        if not data:
+            return Response({'error': 'Data is required'}, status=status.HTTP_400_BAD_REQUEST)
+
+        auth_token = data['authToken']
+        if not auth_token:
+            return Response({'error': 'Auth token is required'}, status=status.HTTP_400_BAD_REQUEST)
+
+        first_name = data['firstName']
+        if not first_name:
+            return Response({'error': 'First name is required'}, status=status.HTTP_400_BAD_REQUEST)
+
+        last_name = data['lastName']
+        if not last_name:
+            return Response({'error': 'Last name is required'}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
It's better and much neater to use serializer here too
you can then use this
```python
serializer = self.get_serializer(data=request.data)
serializer.is_valid(raise_exception=True)
```
This will raise exception with field name, so you wouldn't need to create errors yourself, but the result will differ from what you have now.
Anyway usually it's better to return not only error but also field name etc., so you can easily map response to your forms on FE